client/src/pages/Report/NodesSection/NodesSection.tsx

```diff
@@ -36,9 +36,31 @@ const NodesSection: React.FC<NodesSectionProps> = ({
                                                    }) => {
     const [showModal, setShowModal] = useState(false);
     const [selectedNodes, setSelectedNodes] = useState<NodeDataRow[]>([]);
+    const [availableNodes, setAvailableNodes] = useState<NodeDataRow[]>([]);
     const [showCustomPromptPopup, setShowCustomPromptPopup] = useState(false);
     const [selectedNode, setSelectedNode] = useState<NodeDataRow | null>(null);
 
+    const loadNodes = async () => {
+        try {
+            const data = await ManagmentServiceApiInstance.getNodes(clusterId);
+            setAvailableNodes(
+              data.map((node) => ({
+                  name: node.name,
+                  running: node.running,
+                  accuracy: defaultAccuracy,
+                  customPrompt: '',
+              }))
+            );
+        } catch (error) {
+            console.error('Failed to fetch nodes:', error);
+        }
+    };
+
+    const handleOpenModal = async () => {
+        await loadNodes();
+        setShowModal(true);
```

Comment:
Wouldn't it be better to open the popup first and show loading animation inside of it while the data is being loaded?

Reply:
hmmm good point, can I fix that in next separate PR? I need those changes asap for presentation

Reply:
Sure